Initbuffer

willghatch opened the pull request:
This adds an option to use the current contents of `$BUFFER` as initial search terms.  Closes issue #49.  @termoshtt would you review this?  I feel like I don't want to merge my own pull requests, so that there is some amount of peer review.

termoshtt commented:
I confirm your code works well.
The `-s` option for `filter-select` is good approach.
(You have missed to write the usage of `-s` option at the top of `filter-select`)

I have two questions:

- Why you introduce `current-buffer-zaw`?
- Can you use `options` functionality in sources to enable `-s` option like `options=(-s ${BUFFER})`
  instead of introducing `-b` option in `zaw` itself?

willghatch commented:
Aha, the answer is clearly because I completely forgot about the options on a source when I added the -b option to zaw.  I'll fix that up.

willghatch commented:
Fixed.  Thanks for your feedback.  It's now squashed to one commit.  The extra one is just that my first squash left an extraneous whitespace change in another file, so I pushed another amendment.

termoshtt commented:
Good work! :+1:
